packages/paste-core/components/disclosure/src/utils.ts

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import {GenericThemeShape} from '@twilio-paste/theme';
+import {BoxStyleProps} from '@twilio-paste/box';
+import {HeadingProps} from '@twilio-paste/heading';
+import {ChevronDisclosureExpandedIconProps} from '@twilio-paste/icons/esm/ChevronDisclosureExpandedIcon';
+
+/**
+ * Returns the icon size based on the heading variant
+ * @param {HeadingProps['variant']} variant
+ * @returns {ChevronDisclosureExpandedIconProps} sizeIcon token for each heading variant
+ */
+export const getIconSize = (variant: HeadingProps['variant']): ChevronDisclosureExpandedIconProps['size'] => {
+  switch (variant) {
+    case 'heading10':
+      return 'sizeIcon90';
+    case 'heading20':
+      return 'sizeIcon70';
+    case 'heading30':
+      return 'sizeIcon60';
+    case 'heading40':
+      return 'sizeIcon40';
+    case 'heading50':
+      return 'sizeIcon30';
+    case 'heading60':
+    default:
+      return 'sizeIcon20';
+  }
+};
+
+/**
+ * Returns the styles needed for the icon color and transitions
+ * @param {boolean} isHovered
+ * @param {HeadingProps['variant']} variant
+ * @param {boolean} ariaExapanded
+ * @param {boolean} ariaDisabled If set to true, the function will return only a color style
+ * @param {GenericThemeShape['space']} space
+ * @returns {BoxStyleProps} Color style if ariaDisabled is false, or color, transform, and transition styles depending on variant if ariaDisabled is false
+ */
+export const getIconHoverStyles = (
+  isHovered: boolean,
+  variant: HeadingProps['variant'],
+  ariaExpanded: boolean | 'true' | 'false' | undefined,
+  ariaDisabled: boolean | 'true' | 'false' | undefined,
+  space: GenericThemeShape['space']
+): Pick<BoxStyleProps, 'color' | 'transform' | 'transition'> => {
+  if (!ariaDisabled) {
+    switch (variant) {
```

Comment:
Can be simplified to:

```js
let slideDistance = space.space20;
if (variant === 'heading40' || variant === 'heading50' || variant === 'heading60') {
  slideDistance = space.space10;
}
return {
  color: isHovered ? 'colorText' : 'colorTextIcon',
  transform: `${isHovered ? `translateX(${slideDistance})` : `translateX(0)`} ${
    ariaExpanded ? 'rotate(0deg)' : 'rotate(-90deg)'
  }`,
  transition: 'transform 200ms ease-out',
};
```

Reply:
Good suggestion. That switch statement was a little hard to follow.